src/components/ChannelPreview/ChannelPreview.js

```diff
@@ -4,24 +4,29 @@ import PropTypes from 'prop-types';
 import { ChatContext } from '../../context';
 import { useLatestMessagePreview } from './hooks/useLatestMessagePreview';
 
-const ChannelPreviewWithContext = React.memo((props) => {
-  const { channel, client } = props;
+const ChannelPreview = (props) => {
+  const { channel } = props;
+  const { client } = useContext(ChatContext);
   const [lastMessage, setLastMessage] = useState({});
   const [unread, setUnread] = useState(channel.countUnread());
   const latestMessage = useLatestMessagePreview(channel, lastMessage);
 
   useEffect(() => {
-    const handleNewMessageEvent = (e) => {
+    const handleEvent = (e) => {
       setLastMessage(e.message);
       setUnread(channel.countUnread());
     };
 
-    channel.on('message.new', handleNewMessageEvent);
+    channel.on('message.new', handleEvent);
+    channel.on('message.updated', handleEvent);
```

Comment:
small efficiency but in the `handleEvent` function should we check the message type before setting unread and save that call from executing if it's just this update message event?

Reply:
Makes sense!!

Reply:
Done!!